tests/test_add_model.py

```diff
@@ -0,0 +1,423 @@
+            "a_fixed_point_decimal",
+            "a_datetime",
+            "a_list_of_strings",
+            "a_nested_attribute",
+            "a_list_of_nested_types",
+        ],
+        ["a_string"],
+        [],
+    ),
+    (MockBasicWithSwaggerMetadata1, [], [], []),
+    (MockBasicWithSwaggerMetadata2, [], [], []),
+]
+
+
+@patch("flask_restful_swagger.swagger.registry")
+@pytest.mark.parametrize(
+    "test_input,properties,required,defaults", test_fixtures
+)
+def test_integration_test_add_model(
+    mock_registry, test_input, properties, required, defaults
+):
+    """Integration test for `add_model(...)` method.
+    Ensures models are added to `registry["models"]` with expected structure.
+    Example each model should have 'description', 'id','notes',
+    'properties', etc.
+
+    Example `registry["models"]`:
+        # print(registry["models"])
+        {   'models': {   .....
+              'TodoItem': {   'description': 'This is an example of a '
+                                             'model class that has '
+                                             'parameters in its '
+                                             'constructor',
+                              'id': 'TodoItem',
+                              'notes': 'and the fields in the swagger spec '
+                                       'are derived from the '
+                                       'parameters<br/>to __init__.<br/>In '
+                                       'this case we would have args, arg2 '
+                                       'as required parameters and arg3 '
+                                       'as<br/>optional parameter.',
+                              'properties': {   'arg1': {'type': 'string'},
+                                                'arg2': {'type': 'string'},
+                                                'arg3': {   'default': '123',
+                                                            'type': 'string'}},
+                              'required': ['arg1', 'arg2']},
+                                ..........
+    """
+    _mock_registry = {"models": {}}
+    mock_registry.__getitem__.side_effect = _mock_registry.__getitem__
+    mock_registry.__setitem__.side_effect = _mock_registry.__setitem__
+
+    swagger.add_model(test_input)
+
+    assert test_input.__name__ in _mock_registry["models"]
+    assert "description" in _mock_registry["models"][test_input.__name__]
+    assert "notes" in _mock_registry["models"][test_input.__name__]
+
+    if "resource_fields" in dir(test_input):
+        if hasattr(test_input, "required"):
+            assert "required" in _mock_registry["models"][test_input.__name__]
+    else:
+        assert "required" in _mock_registry["models"][test_input.__name__]
+
+    assert "properties" in _mock_registry["models"][test_input.__name__]
+
+
+# Setup test - ensure `_parse_doc(..)` is called without issues
+test_fixtures_model_inputs = [
```

Comment:
There are a number of data structures sort of sprinkled in.  what are you thoughts?  Is it better to include them in line so you can see the data inputs, or better to import them to reduce the size of this file?  I'm unsure what's better for organizing this very well written, but long test.

Reply:
Refactored the data structure setup into seperate file `tests/fixtures_add_model.py`.  Using context managers to centralise the setup.